code/modules/ventcrawl/ventcrawl.dm

```diff
@@ -0,0 +1,223 @@
+var/list/ventcrawl_machinery = list(
+	/obj/machinery/atmospherics/unary/vent_pump,
+	/obj/machinery/atmospherics/unary/vent_scrubber
+	)
+
+// Vent crawling whitelisted items, whoo
+/mob/living/var/list/can_enter_vent_with = list(
+	/obj/item/weapon/implant,
+	/obj/item/device/radio/borg,
+	/obj/item/weapon/holder,
+	/obj/machinery/camera,
+	/mob/living/simple_animal/borer
+	)
+
+/mob/living/var/list/icon/pipes_shown = list()
+/mob/living/var/last_played_vent
+/mob/living/var/is_ventcrawling = 0
+/mob/var/next_play_vent = 0
+
+/mob/living/proc/can_ventcrawl()
+	return 0
+
+/mob/living/Login()
+	. = ..()
+	//login during ventcrawl
+	if(is_ventcrawling && istype(loc, /obj/machinery/atmospherics)) //attach us back into the pipes
+		remove_ventcrawl()
+		add_ventcrawl(loc)
+
+/mob/living/carbon/slime/can_ventcrawl()
+	if(Victim)
+		src << "<span class='warning'>You cannot ventcrawl while feeding.</span>"
+		return 0
+	return 1
+
+/mob/living/proc/is_allowed_vent_crawl_item(var/obj/item/carried_item)
+	for(var/type in can_enter_vent_with)
+		if(istype(carried_item, can_enter_vent_with))
+			return get_inventory_slot(carried_item) == 0
+	return 0
+
+/mob/living/carbon/is_allowed_vent_crawl_item(var/obj/item/carried_item)
+	if(carried_item in internal_organs)
+		return 1
+	return ..()
+
+/mob/living/carbon/human/is_allowed_vent_crawl_item(var/obj/item/carried_item)
```

Comment:
This is unnecessary. If the parent (on line 42) is defined the same way, then there's no point in redefining the child as well. It inherits fully already.